crates/ruff_linter/src/rules/ruff/rules/unused_async.rs

```diff
@@ -0,0 +1,169 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::identifier::Identifier;
+use ruff_python_ast::visitor::preorder;
+use ruff_python_ast::{self as ast, AnyNodeRef, Expr, Stmt};
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for functions declared `async` that do not await or otherwise use features requiring the
+/// function to be declared `async`.
+///
+/// ## Why is this bad?
+/// Declaring a function `async` when it's not is usually a mistake, and will artificially limit the
+/// contexts where that function may be called. In some cases, labeling a function `async` is
+/// semantically meaningful (e.g. with the trio library).
+///
+/// ## Examples
+/// ```python
+/// async def foo():
+///     bar()
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def foo():
+///     bar()
+/// ```
+#[violation]
+pub struct UnusedAsync {
+    name: String,
+}
+
+impl Violation for UnusedAsync {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let UnusedAsync { name } = self;
+        format!("Function `{name}` is declared `async`, but doesn't await or use async features.")
```

Comment:
Nit: wrap await in backticks here for consistency with async?

Reply:
Implemented below.